command/ca/token.go

```diff
@@ -295,6 +323,31 @@ func tokenAction(ctx *cli.Context) error {
 		}
 	}
 
+	// Add options to create a confirmation claim if a CSR or SSH public key is
+	// passed.
+	var tokenOpts []cautils.Option
+	if cnfFile != "" {
+		in, err := utils.ReadFile(cnfFile)
+		if err != nil {
+			return err
+		}
+		if isSSH {
+			sshPub, _, _, _, err := ssh.ParseAuthorizedKey(in)
+			if err != nil {
+				return errors.Wrap(err, "error parsing ssh public key")
+			}
+			tokenOpts = append(tokenOpts, cautils.WithSSHPublicKey(sshPub))
+		} else {
+			csr, err := pemutil.ParseCertificateRequest(in)
+			if err != nil {
+				return errors.Wrap(err, "error parsing certificate request")
+			}
+			tokenOpts = append(tokenOpts, cautils.WithCertificateRequest(csr))
```

Comment:
Instead of passing the CSR or public key, the claim itself could be calculated here, and just pass it as the fingerprint directly? Or is there some flow in which to defer calculation of the fingerprint is necessary? `step ca sign`, maybe?

Reply:
`step ca sign` use `cautils.WithCertificateRequest(csr)` and `step ssh certificate` use `cautils.WithSSHPublicKey(sshPub)`, although the SSH one is currently totally ignored. These fields go to the shared context of the `cautils` package and end up in the `token` package, which takes care of calculating the fingerprint and adding the proper attribute to the final token. Right now, only the CSR gets into the token package, but sending the type allows us to specify the right attribute now that we don't share just `kid`.